### PR TITLE
Update AdminController.Edit() to use the new OpenIddictApplicationManager.UpdateAsync() overload

### PR DESCRIPTION
**Pull request by PinpointTownes**
Fixes https://github.com/OrchardCMS/Orchard2/issues/609.

### CODE REVIEW

File: src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs [hunk 1]
```diff
@@ -128,8 +127,11 @@ public class AdminController : Controller
         {
             if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageOpenIdApplications))
                 return Unauthorized();
-            
-            if (model.UpdateClientSecret && string.IsNullOrWhiteSpace(model.ClientSecret))
+
+            if (model.Type == ClientType.Public && !string.IsNullOrEmpty(model.ClientSecret))
+                ModelState.AddModelError(nameof(model.ClientSecret), "No client secret can be set for public applications.");
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs]: Please translate using `IStringLocalizer`. The same way you fix bad indentation feel free to fix bad localization. Same two lines below.

File: src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs [hunk 2]
```diff
@@ -128,8 +127,11 @@ public class AdminController : Controller
...
+
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs]: Between @jersiovic 's and your code, I'd like to remind you two that we try to add curly braces on all if/else statements, even one-liners, and not add an empty line between `if` and `else`.

**Comment by sebastienros** [comment]
BTW @PinpointTownes  feel free to create your own branches in this repos, that lets us participate on it sometimes.

File: src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs (hunk 1 above)
Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs]: Fixed, thanks.

File: src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs (hunk 2 above)
Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs]: I always use braces for if/else if/else states, but I followed the same code style for consistency here.

**Comment by PinpointTownes** [comment]
> BTW @PinpointTownes feel free to create your own branches in this repos, that lets us participate on it sometimes.

I never do that for minor changes because it adds way too much noise in the git history (and hem, [Orchard has already a bunch of totally stale branches](https://github.com/OrchardCMS/Orchard2/branches/stale)). 

Note that users with write access to the main repo can push new commits to forked repos, so nothing prevents you from doing that with my forked branch :smile:

**Comment by sebastienros** [comment]
> Note that users with write access to the main repo can push new commits to forked repos

I had no idea

**Comment by PinpointTownes** [comment]
@sebastienros

image

**Comment by sebastienros** [comment]
It seems to be a relatively new feature: https://github.com/blog/2247-improving-collaboration-with-forks
That's great, I won't bother people as much from now on.

**Pull request merged**